src/odemis/driver/hamamatsurx_simcam.py

```diff
@@ -0,0 +1,552 @@
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with Odemis. If not, see http://www.gnu.org/licenses/.
+'''
+
+from __future__ import division
+
+import Queue
+import logging
+from odemis import model, util, dataio
+from odemis.model import oneway
+import time
+import numpy
+import os
+
+
+class ReadoutCamera(model.DigitalCamera):
+    """
+    Represents Hamamatsu readout camera.
+    """
+
+    def __init__(self, name, role, parent, spectrograph=None, daemon=None, **kwargs):
+        """ Initializes a fake Hamamatsu OrcaFlash readout camera.
+        :parameter name: (str) as in Odemis
+        :parameter role: (str) as in Odemis
+        :parameter parent: class streakcamera
+        """
+        # TODO image focus and operate mode
+        # get the fake images
+        try:
+            image = kwargs.pop("image")
+            image = unicode(image)
+            # ensure relative path is from this file
+            if not os.path.isabs(image):
+                image = os.path.join(os.path.dirname(__file__), image)
+            converter = dataio.find_fittest_converter(image, mode=os.O_RDONLY)
+            self._img = []
+            for i in range(10):
+                img = converter.read_data(image)[i+2]
+                if img.ndim > 3:  # remove dims of length 1
+                    img = numpy.squeeze(img)
+                self._img.append(img)  # can be RGB or greyscale
+            self._img_counter = 0  # image counter to provide different images to live view
+        except:
+            raise Exception("No fake image provided")
+
+        super(ReadoutCamera, self).__init__(name, role, parent=parent,
+                                            daemon=daemon, **kwargs)  # init HwComponent
+
+        self.parent = parent
+
+        self._spectrograph = spectrograph
+        if not spectrograph:
+            logging.warning("No spectrograph specified. No wavelength metadata will be attached.")
+
+        self._metadata[model.MD_HW_VERSION] = 'Simulated readout camera OrcaFlash 4.0 V3, ' \
+                                              'Product number: C13440-20C, Serial number: 301730'
+        self._metadata[model.MD_SW_VERSION] = 'Firmware: 4.20.B, Version: 4.20.B03-A19-B02-4.02'
+        self._metadata[model.MD_DET_TYPE] = model.MD_DT_INTEGRATING
+
+        # sensor size (resolution)
+        # Note: sensor size of OrcaFlash is actually much larger (2048px x 2048px)
+        # However, only a smaller subarea is used for operating the streak system.
+        resolution = (1344, 1016)  # x (lambda): horizontal, y (time): vertical
```

Comment:
Should be based on the `_image.shape`?

Reply:
How do I get the binning info from the MD when reading the image.
self._img[0].metadata does not seem to include any exposure time or binning MD